### PR TITLE
feat(connector): [PowerTranz] Use connector_response_reference_id as reference to merchant

### LINKED ISSUE #2345: [FEATURE]: [PowerTranz] Use `connector_response_reference_id` as reference to merchant
### :memo: Feature Description

- Reference id are used to map transactions in the connector’s dashboard.
- Hyperswitch manages several reference ids, such as `payment_id`, `attempt_id`, and `connector_transaction_id` for a single transaction. 
- However, merchants may encounter uncertainty when determining which ID to utilize in the connector dashboard to identify the payment.

### :hammer: Possible Implementation

- When we receive a response from the connector for the payment, we deserialize it and populate the `response` field in RouterData. 
- For the `TransactionResponse` type, we must fill the `connector_response_reference_id` with a corresponding reference id for the merchant to identify the transaction. 
- One might need to have exposure to api docs of the connector for which it is being implemented to decide what to fill in connector_response_reference_id.
- You can check this PR for further reference https://github.com/juspay/hyperswitch/pull/1735

🔖 Note: All the changes needed should be contained within `hyperswitch/crates/router/src/connector/`

### :package: Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### :package: Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by kritikash18**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

Fixes #2345 

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/powertranz/transformers.rs [hunk 1]
```diff
@@ -239,6 +241,7 @@ pub struct PowertranzBaseResponse {
     iso_response_code: String,
     redirect_data: Option<String>,
     response_message: String,
+    reference: Option<String>,
```
Comment (srujanchikke) [inline crates/router/src/connector/powertranz/transformers.rs:244]: Same in response , the merchat reference would be order_identifier . 
```suggestion
    order_identifier : Option<String>,
```

File: crates/router/src/connector/powertranz/transformers.rs [hunk 2]
```diff
@@ -29,6 +29,7 @@ pub struct PowertranzPaymentsRequest {
     // billing_address: Option<PowertranzAddressDetails>,
     // shipping_address: Option<PowertranzAddressDetails>,
     extended_data: Option<ExtendedData>,
+    reference: String,
```
Comment (srujanchikke) [inline crates/router/src/connector/powertranz/transformers.rs:32]: Hi @kritikash18 ,

In reference to issue #2345 , we need to adjust the reference field mapping based on the API documentation of the connector. `order_identifier` is merchant reference id for powertranz. 

the api documentation for powertranz is attached below
[PowertranzNon3DSAPI-V-2.0.pdf](https://github.com/juspay/hyperswitch/files/12789748/PowertranzNon3DSAPI-V-2.0.pdf)

Comment (kritikash18) [reply crates/router/src/connector/powertranz/transformers.rs]: updated

File: crates/router/src/connector/powertranz/transformers.rs (hunk 1 above)
Comment (kritikash18) [reply crates/router/src/connector/powertranz/transformers.rs]: done

File: crates/router/src/connector/powertranz/transformers.rs (hunk 2 above)
Comment (srujanchikke) [reply crates/router/src/connector/powertranz/transformers.rs]: Currently `order_identifier` is mapped to `payment_id` , it should be updated to `connector_request_reference_id` in payment request try_from .

Comment (kritikash18) [reply crates/router/src/connector/powertranz/transformers.rs]: updated; I'm new to rust so apologies for the extra review effort 😅

**Comment by srujanchikke** [comment]
Looks good to me!

Please update the branch to ensure the spell check in the CI workflow functions properly.

**Review by srujanchikke — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @kritikash18!

**Comment by VedantKhairnar** [comment]
Hey @kritikash18  :wave:,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.
Thanks!

**Pull request merged**